torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -588,6 +615,201 @@ def get_detector_name(self) -> str:
 
+    def _extract_input_info(self, model: GraphModule) -> Dict[str, Dict]:
+        r"""
+        Takes in a callibrated GraphModule and then finds the relavent observers.
+        It then extracts the input information for each observer returns it
+
+        Args
+            model (GraphModule): The prepared and calibrated GraphModule with inserted ModelReportObservers
+
+        Returns a dict mapping relavent module fqns (str) to a dict with keys:
+            "per_channel_max" : maps to the per_channel max values
+            "per_channel_min" : maps to the per_channel min values
+            "global_max" : maps to the global max recorded
+            "global_min" : maps to the global min recorded
+        """
+
+        # return dictionary mapping observer fqns to desired info
+        input_info: Dict[str, Dict] = {}
+
+        for fqn, module in model.named_modules():
+            # check to see if module is of a supported type
+            is_supported_type: bool = (
+                sum(list(map(lambda x: type(module) is x, self.SUPPORTED_MODULES))) > 0
+            )
+
+            # if module is supported and it has a pre-observer
+            if is_supported_type and hasattr(module, self.DEFAULT_PRE_OBSERVER_NAME):
+                # get pre observer for the module
+                pre_obs = getattr(module, self.DEFAULT_PRE_OBSERVER_NAME)
+
+                input_info[fqn] = {
+                    self.PER_CHANNEL_MAX_KEY: pre_obs.max_val,
+                    self.PER_CHANNEL_MIN_KEY: pre_obs.min_val,
+                    self.GLOBAL_MAX_KEY: max(pre_obs.max_val),
+                    self.GLOBAL_MIN_KEY: min(pre_obs.min_val),
+                }
+
+        return input_info
+
+    def _extract_weight_info(self, model: GraphModule) -> Dict[str, Dict]:
+        r"""
+        Takes in a callibrated GraphModule and then finds the relavent observers.
+        It then extracts the weight information for each layer an observer is attached to.
+
+        Args
+            model (GraphModule): The prepared and calibrated GraphModule with inserted ModelReportObservers
+
+        Returns a dict mapping module fqns (str) to a dict with keys:
+            "per_channel_max" : maps to the per_channel max values
+            "per_channel_min" : maps to the per_channel min values
+            "global_max" : maps to the global max recorded
+            "global_min" : maps to the global min recorded
+        """
+        # return dictionary mapping observer fqns to desired info
+        weight_info: Dict[str, Dict] = {}
+
+        for fqn, module in model.named_modules():
+            # check to see if module is of a supported type
+            is_supported_type: bool = (
+                sum(list(map(lambda x: type(module) is x, self.SUPPORTED_MODULES))) > 0
+            )
+
+            # if module is supported and it has a pre-observer
+            if is_supported_type and hasattr(module, self.DEFAULT_PRE_OBSERVER_NAME):
```

Comment:
I'm seeing this structure repeated in a lot of places. I wonder if all detector classes should just define a `self.get_supported_modules(model)` method to minimize duplication